metashare/api/views.py

```diff
@@ -261,3 +261,16 @@ def redirect(self, request, pk=None):
             )
         url = scratch_org.get_login_url()
         return HttpResponseRedirect(redirect_to=url)
+
+    @action(detail=True, methods=["POST"])
+    def refresh(self, request, pk=None):
+        scratch_org = self.get_object()
+        if not request.user == scratch_org.owner:
+            return Response(
+                {"error": _("Requesting user did not create scratch org.")},
+                status=status.HTTP_403_FORBIDDEN,
+            )
```

Comment:
@wlonk Because this eventually creates a new org, which requires `user.sf_username`, this needs to also check `if self.request.user.is_devhub_enabled`, similar to the `perform_create` action above.

Actually, maybe better: the job can store off the `scratch_org.owner_sf_username` and use that in the re-creation instead of `user.sf_username`.

Reply:
@wlonk Bump on this.

Reply:
Ah, correct. Will fix.

Reply:
@jgerigmeyer I went with the smaller approach, as the "maybe better" would require extensive changes to the current jobs. We could make a new story to investigate that, though.